cirq-core/cirq/circuits/circuit_dag.py

```diff
@@ -92,28 +98,49 @@ def __init__(
         """
         super().__init__(incoming_graph_data)
         self.can_reorder = can_reorder
-        self.device = device
+        self._device = device
+
+    @property  # type: ignore
+    @_compat.deprecated(
+        deadline='v0.15',
+        fix=circuit._DEVICE_DEP_MESSAGE,
+    )
+    def device(self) -> devices.Device:
+        return self._device
 
     @staticmethod
     def make_node(op: 'cirq.Operation') -> Unique:
         return Unique(op)
 
     @staticmethod
     def from_circuit(
-        circuit: 'cirq.Circuit',
+        qcircuit: circuit.Circuit,
```

Comment:
Why change the parameter name?  Isn't this backwards incompatible?

Reply:
Reverted.